hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactingToCellFlatMapMemStore.java

```diff
@@ -903,6 +904,37 @@ public void testForceCopyOfBigCellIntoImmutableSegment() throws IOException {
     }
   }
 
+  /**
+   * Test big cell size after in memory compaction. (HBASE-26467)
+   * @throws IOException
+   */
+  @Test
+  public void testBigCellSizeAfterInMemoryCompaction() throws IOException {
+    MemoryCompactionPolicy compactionType = MemoryCompactionPolicy.BASIC;
+    memstore.getConfiguration().setInt(MemStoreCompactionStrategy
+      .COMPACTING_MEMSTORE_THRESHOLD_KEY, 1);
+    memstore.getConfiguration().set(CompactingMemStore.COMPACTING_MEMSTORE_TYPE_KEY,
+      String.valueOf(compactionType));
+    ((MyCompactingMemStore) memstore).initiateType(compactionType, memstore.getConfiguration());
+
+    byte[] val = new byte[MemStoreLAB.CHUNK_SIZE_DEFAULT];
+
+    long size = addRowsByKeys(memstore, new String[]{"A"}, val);
+    ((MyCompactingMemStore) memstore).flushInMemory();
+
+    while (((CompactingMemStore) memstore).isMemStoreFlushingInMemory()) {
```

Comment:
((MyCompactingMemStore) memstore).flushInMemory() is a synchronous method,seems here is no need to wait the flush completed.